proto/cosmos/staking/v1beta1/tx.proto

```diff
@@ -120,3 +124,18 @@ message MsgUndelegate {
 message MsgUndelegateResponse {
   google.protobuf.Timestamp completion_time = 1 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
 }
+
+// MsgCancelUnbondingDelegation defines the SDK message for performing an cancel unbonding delegation for delegator
+message MsgCancelUnbondingDelegation{
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string                   delegator_address = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  string                   validator_address = 2 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  cosmos.base.v1beta1.Coin amount            = 3 [(gogoproto.nullable) = false];
+  // creation_height is the height which the unbonding took place.
+  uint64 creation_height = 4;
```

Comment:
Is this solely to cancel/remove a specific existing unbonding entry? Do we absolutely need this? I'm just thinking from a UX POV, it's somewhat...not great lol.

Just thinking out loud here, what if we just have `amount`? And then the message handler iterates through each entry summing up the total and once total >= `amount`, we remove all those entries? Granted, this means the user might have slightly more unbonded than they requested.

Reply:
This is for specific existing unbonding entry  (delegators can have multiple unbonding delegations for one validator)
delegators needs to enter how much amount wants to re-delegate back to validator from ubd entry
 
 ```go
amount  = msg.Amount 
if !udbEntry.Balance.Lt(amount){
   // handle error throws invalid amount 
}else{
 // handle logic for remove or update the udb entry 
 // and delegate back to validator 
}
```

Reply:
@alexanderbez  I updated pr description for more info

Reply:
Yes , we will show one button for re-delegate on each ubd entry row in UI, 
on background clients know exact height and when user click on button one pop up will appear , user will enter how much wants to re-delegate from ubd entry. 
```json 
{
  "unbonding_responses": [
    {
      "delegator_address": "string",
      "validator_address": "string",
      "entries": [
        {
          "creation_height": "string",
          "completion_time": "2022-01-16T15:26:15.278Z",
          "initial_balance": "string",
          "balance": "string"
        }
      ]
    }
  ],
  "pagination": {
    "next_key": "string",
    "total": "string"
  }
}
```

Reply:
Ok, that makes sense if the UI can handle it and knows the height for each entry. I still don't think it's the cleanest UX, but it makes sense. I will review the rest of the PR shortly 👍

Reply:
Thanks